Skyscanner/backpack
# [BPK-1682] videos on docs site

The original card was to add the Alert component video on the docsite but given this PR will be quite big I don't want to add too much noise and have the focus around the important files.

some if not most of the code has been ~stolen~ borrowed from the PR we made at the designer offsite (https://github.com/Skyscanner/backpack/compare/story-control?expand=1).

One important thing to notice is the different platforms recorders, the iOS one is pretty straightforward and because I need the same API to work seemlessly on testing iOS and Android the android one has some code stolen from my repo https://github.com/matteo-hertel/android-commander/blob/master/src/platforms/android.js

*Note* the `adb` cli is needed for the android recording to work, to check if you have it installed run
`adb --version` if it fails, install the cli with `brew cask install android-platform-tools` (thanks @shaundon for spotting this)

To test this:

- run `npm install` or `npm run clean && npm i` if you're brave or you need a ☕️
  - FYI this will add `puppeteer` which is roughly 70MB 
- run the native storybook `npm run native`
- run the detox as before `npm run detox:test:android` (or `npm run detox:test:ios` if you're into it)
- everything should succeed and you should not see any kind of error

once the above is done, to further prove your awesomeness and thoroughly test the whole thing you can run from the root
- `rm  native/packages/react-native-bpk-component-switch/videos/android/*` or `rm  native/packages/react-native-bpk-component-switch/videos/ios/*`
- re-run the test above
- the video files should be back in place

The docs site can be tested by running `npm run docs` and once that is done visit (http://0.0.0.0:8080/components/switch?platform=native)[http://0.0.0.0:8080/components/switch?platform=native]

one maim thing to notice is the usage of `sleep(time)` across the board, all the times are made up (trying to be sensible) and they can potentially be changed if needed
